app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_worker/parser.rb

```diff
@@ -0,0 +1,80 @@
+class ManageIQ::Providers::AnsibleTower::ConfigurationManager::RefreshWorker::Parser
+  attr_accessor :inventory, :ems
+
+  def initialize(ems, target, inventory)
+    @ems    = ems
+    @target = target
+    @inventory = inventory
+  end
+
+  def parse
+    [
+      inventory_groups,
+      configured_systems,
+      configuration_scripts,
+    ]
+  end
+
+  def inventory_groups
+    # FIXME: memoization  should not be needed.
+    @inventory_groups ||= ManagerRefresh::InventoryCollection.new(
+      ManageIQ::Providers::ConfigurationManager::InventoryRootGroup,
+      :association => :inventory_root_groups,
+      :parent      => ems,
+    ).tap do |c|
+      inventory.inventories.each do |i|
+        c << c.new_inventory_object(
+          # to_s should not be necessary, but its needed to resolve lazy find
+          :ems_ref => i.id.to_s,
+          :manager => ems,
+          :name    => i.name,
+        )
+      end
+    end
+  end
+
+  def configured_systems
+    ManagerRefresh::InventoryCollection.new(
+      ManageIQ::Providers::AnsibleTower::ConfigurationManager::ConfiguredSystem,
+      :association => :configured_systems,
+      # FIXME: array should not be needed
+      # FIXME: manager_ref should not be needed here
+      :manager_ref => [:manager_ref],
```

Comment:
I think you need to define this. In ideal world, there would be a DB unique index here, then we can take manager_ref from that.